### PR TITLE
Move RubyLauncher to our repository

### PR DESCRIPTION
**Pull request by pitr-ch**
Do not rebase

### CODE REVIEW

File: bin/truffleruby.sh [hunk 1]
```diff
@@ -92,14 +92,16 @@ source_truffle="$root_parent/graal/truffle/mxbuild"
...
 elif [ -f "$source_truffle/dists/truffle-api.jar" ]; then # Source Truffle suite
     truffle="$source_truffle"
     graal_sdk="$root_parent/graal/sdk/mxbuild/dists/graal-sdk.jar"
+    launcher_common="$root_parent/graal/sdk/mxbuild/dists/launcher-common.jar"
 else
     echo "Could not find Truffle jars" 1>&2
     exit 1
 fi
-java_args+=("-Xbootclasspath/a:$truffle/dists/truffle-api.jar:$graal_sdk")
+java_args+=("-Xbootclasspath/a:$truffle/dists/truffle-api.jar:$graal_sdk:$launcher_common")
```
Comment (eregon) [inline bin/truffleruby.sh]: Does `launcher-common.jar` need to be on the bootclasspath or is it enough on the classpath?

File: src/launcher/java/org/truffleruby/launcher/RubyLauncher.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2017, 2017, Oracle and/or its affiliates. All rights reserved.
+ * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
+ */
```
Comment (eregon) [inline src/launcher/java/org/truffleruby/launcher/RubyLauncher.java]: The copyright should be adapted.

Comment (pitr-ch) [reply src/launcher/java/org/truffleruby/launcher/RubyLauncher.java]: fixed

File: bin/truffleruby.sh (hunk 1 above)
Comment (pitr-ch) [reply bin/truffleruby.sh]: I think bootpath. I am not 100% sure, but I don't think it makes sense to go and find out. This is a developer launcher which is not distributed. I don't think there is any practical impact of the placement.

Comment (eregon) [reply bin/truffleruby.sh]: Right, fair enough and the rest of Graal SDK is there too.
Note that this launcher is notably used for the testing distribution and distributions without a SVM image launcher.

File: src/launcher/java/org/truffleruby/launcher/RubyLauncher.java (hunk 2 above)
Comment (eregon) [reply src/launcher/java/org/truffleruby/launcher/RubyLauncher.java]: I meant the copyright text/license is wrong (should be our usual tri-license).

File: src/launcher/java/org/truffleruby/launcher/RubyLauncher.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2017, 2018, Oracle and/or its affiliates. All rights reserved.
+ * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
+ */
```
Comment (chrisseaton) [inline src/launcher/java/org/truffleruby/launcher/RubyLauncher.java]: Copyright, as Benoit said.

**Pull request merged**